### DISCUSSION TITLE
Landscape tables

### DISCUSSION THREAD

**Discussion by sergarcia** [ickc/pantable#33] [Q&A]
I am trying to include a big table from a file in my markdown document, which I want to appear in a landscape (the rest of the document uses portrait) using [pdflscape](https://ctan.org/pkg/pdflscape). I am having  a couple of issues.
1- The output is not correct, a string of YAML appears where the table should be (without landscape the table works fine).
2- It takes really long to compile the document.

Is there a better way to make big tables appear in landscape?

EDIT: I solved the problem by including a tag in the the .md file and then converting to .tex, and then replacing the tag by the appropriate LaTeX syntax. My  compilation script currently looks like this:
```bash
#!/bin/sh
FILE_NAME="report"
pandoc -s -N -F pantable "$FILE_NAME.md" -o "$FILE_NAME.tex"
sed -i 's/begin{figure}\[htbp\]/begin{figure}\[H\]/g' "$FILE_NAME.tex"
sed -i 's/!beginlandscape/\\begin{landscape}/g' "$FILE_NAME.tex"
sed -i 's/!endlandscape/\\end{landscape}/g' "$FILE_NAME.tex"
pdflatex "$FILE_NAME.tex" -0 "$FILE_NAME.pdf" > /dev/null
rm "$FILE_NAME.tex"
rm "$FILE_NAME.log"
rm "$FILE_NAME.out"
rm "$FILE_NAME.aux"
```

**Comment by ickc** [comment]
can you provide a minimum working example? (including the source, command used, pandoc version, platform, etc.)

Regarding performance: if you turn on markdown support in csv table, each cell is parsed from markdown to native individually, and each time it called pandoc once. So this is an overhead of calling an externel executables (the starup time there.) This limitation comes from the fact that "there's no real pandoc library in Python", by this I mean all existing Python pandoc library actually calls the pandoc cli program with a bunch of commands, rather than e.g. using the C FFI and build a proper C module calling Haskell library.

I've an idea to overcome this limiation, but that would requires me having some free time to sit down to experiment with it and implement it. Let me know if you're interested to implement it and I'll share my idea.

**Comment by ickc** [comment]
I suspect you used `\begin...\end...` syntax to enclose the table? If so, anything between the begin and end will be interpreted as pure LaTeX and hence it won't be processed by the filter.

Since there aren't any MWE to work with, closing it now.
